### PR TITLE
DOP-4672 - Allows the background styles to be influenced by the dark mode value from it's context

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-4672

### Current Behavior:

[Atlas](https://www.mongodb.com/docs/atlas/)

### Staging Links:

[Stagged Atlas](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/caesarbell/DOP-4672/index.html)

### Notes:

- Body is now influenced by the mode selected, i.e. when a user clicks the toggle button, the new styles for the background color and text will be applied.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by seungpark — COMMENTED** [review]
this LGTM, as far as changes independent from DOP-4671. minor comments below but overall looks great!

**Review by mmeigs — COMMENTED** [review]
Looks great! 
Two overall nits: 
- Make sure not to merge until you've first merged in the change I asked for in [DOP-4667 PR](https://github.com/mongodb/snooty/pull/1097)
- Remember to add context to the title of the PR

File: src/styles/mongodb-docs.css [hunk 1]
```diff
@@ -484,6 +484,7 @@ th {
   font-feature-settings: normal;
 }
 :root {
+  --gray-dark4: #001E2B;
```
[the remark this answers is not in the document]
Comment (caesarbell) [reply src/styles/mongodb-docs.css:487]: @seungpark can we circle back on this after the PR for the `Steps` Component?

[the remark this answers is not in the document]
Comment (seungpark) [reply src/styles/mongodb-docs.css:487]: sure we can circle back to this if we need to prioritize SSG'ing dark mode (without a flash of darkMode=false default for LeafyGreenProvider)

File: src/styles/global-dark-mode.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.dark-theme body {
+     background-color: var(--lg-bg-dark-mode);
+     color: var(--lg-color-dark-mode);
+}
+
+.light-theme body {
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/styles/global-dark-mode.css:6]: @caesarbell it was for clarity of styling but not absolutely required. is this already defined in above mentioned stylesheet? did not come across it

[the remark this answers is not in the document]
Comment (caesarbell) [reply src/styles/global-dark-mode.css:6]: @seungpark that is correct, the background and color are already defined in the `mongodb-docs.css` the background-color is defined on line 268, and the color is defined on line 1012.

**Review by mmeigs — APPROVED** [review]

**Pull request merged**